src/common/k8s-api/kube-api.ts

```diff
@@ -436,10 +458,10 @@ export class KubeApi<T extends KubeObject> {
     throw new Error(`GET multiple request to ${url} returned not an array: ${JSON.stringify(parsed)}`);
   }
 
-  async get({ name = "", namespace = "default" } = {}, query?: IKubeApiQueryParams): Promise<T | null> {
+  async get(desc: ResourceDescriptor, query?: IKubeApiQueryParams): Promise<T | null> {
```

Comment:
While this is technically a breaking change I think it is acceptable because a call to `get()` without this first argument would return the "list" result which would then trigger the `Array.isArray()` check on line 468 bellow.